Mathlib/CategoryTheory/MorphismProperty/Lifting.lean

```diff
@@ -0,0 +1,143 @@
+      by rw [Category.assoc, r_w, ← Category.assoc, CommSq.fac_right, Category.assoc, retract_right,
+        Category.comp_id]⟩
+
+lemma llp.IsStableUnderRetracts {T : MorphismProperty C} : T.llp.IsStableUnderRetracts :=
+  ⟨fun {_ _} _ _ _ _ h hg _ _ _ hf ↦ liftOfRetractLeft h (hg hf)⟩
+
+lemma rlp.IsStableUnderRetracts {T : MorphismProperty C} : T.rlp.IsStableUnderRetracts :=
+  ⟨fun {_ _} _ _ _ _ h hf _ _ _ hg ↦ liftOfRetractRight h (hf hg)⟩
+
+open IsPushout in
+instance llp.IsStableUnderCobaseChange {T : MorphismProperty C} :
+    T.llp.IsStableUnderCobaseChange where
+  of_isPushout {_ _ _ _ f s _ t} P hf _ _ g hg := {
+    sq_hasLift := fun {u v} sq ↦ by
+      have w : (s ≫ u) ≫ g = f ≫ (t ≫ v) := by
+        rw [← Category.assoc, ← P.toCommSq.w, Category.assoc, Category.assoc, sq.w]
+      exact ⟨P.desc u ((hf hg).lift ⟨w⟩) (by rw [CommSq.fac_left]), P.inl_desc ..,
+        P.hom_ext (by rw [inl_desc_assoc, sq.w]) (by rw [inr_desc_assoc, CommSq.fac_right])⟩}
+
+open IsPullback in
+instance rlp.IsStableUnderBaseChange {T : MorphismProperty C} :
+    T.rlp.IsStableUnderBaseChange where
+  of_isPullback {_ _ _ _ t f s _} P hf _ _ g hg := {
+    sq_hasLift := fun {u v} sq ↦ by
+      have w : (u ≫ s) ≫ f = g ≫ v ≫ t := by
+        rw [Category.assoc, P.toCommSq.w, ← Category.assoc, ← Category.assoc, sq.w]
+      exact ⟨P.lift ((hf hg).lift (.mk w)) v (by rw [CommSq.fac_right]),
+        P.hom_ext (by rw [Category.assoc, lift_fst, CommSq.fac_left])
+          (by rw [Category.assoc, lift_snd, sq.w]), P.lift_snd _ _ _⟩}
+
+instance llp.ContainsIdentities (T : MorphismProperty C) : T.llp.ContainsIdentities :=
+  ⟨fun _ _ _ _ _ ↦
+    ⟨fun {f _} sq ↦ ⟨f, Category.id_comp _, by simp only [sq.w, Category.id_comp]⟩⟩⟩
+
+instance rlp.ContainsIdentities (T : MorphismProperty C) : T.rlp.ContainsIdentities :=
+  ⟨fun _ _ _ _ _ ↦
+    ⟨fun {_ g} sq ↦ ⟨g, by simp only [sq.w.symm, Category.comp_id], Category.comp_id _⟩⟩⟩
+
+instance llp.IsStableUnderComposition (T : MorphismProperty C) :
+    T.llp.IsStableUnderComposition where
+  comp_mem p q hp hq _ _ i hi := {
+    sq_hasLift := fun {f g} sq ↦
+      ⟨(hq hi).lift ⟨(hp hi).fac_right ⟨show f ≫ i = p ≫ q ≫ g by rw [← Category.assoc, ← sq.w]⟩⟩,
+        by aesop, (hq hi).fac_right _⟩}
+
+instance rlp.IsStableUnderComposition (T : MorphismProperty C) :
+    T.rlp.IsStableUnderComposition where
+  comp_mem p q hp hq _ _ i hi := {
+    sq_hasLift := fun {f g} sq ↦
+      ⟨(hp hi).lift ⟨((hq hi).fac_left ⟨show (f ≫ p) ≫ q = i ≫ g
+        by rw [Category.assoc, sq.w]⟩).symm⟩, (hp hi).fac_left _, by aesop⟩}
+
+instance llp.IsMultiplicative (T : MorphismProperty C) : T.llp.IsMultiplicative where
+  id_mem := T.llp.id_mem
+  comp_mem := T.llp.comp_mem
+
+instance rlp.IsMultiplicative (T : MorphismProperty C) : T.rlp.IsMultiplicative where
+  id_mem := T.rlp.id_mem
+  comp_mem := T.rlp.comp_mem
+
+instance llp.RespectsIso (T : MorphismProperty C) : T.llp.RespectsIso where
```

Comment:
It would be easier to deduce this from the fact that `llp` is multiplicative and contains isomorphisms.

Reply:
Turns out it's inferred from `IsStableUnderBaseChange` anyway.